the-collab-lab/tcl-79-smart-shopping-list
# Merge dev branch into main

## Description

In this PR, we have integrated Tailwind CSS and ShadCN into the project for styling, aligning the implementation with the Figma design. Key changes include:

- Added modals and hover states for improved user interaction and feedback.
- Enhanced general feedback mechanisms for user interactions with the app.
- Refactored files to ensure styling consistency with the Figma design.
- Added additional code improvements

Note:
While we've made significant progress in implementing the styling, the overall design still requires additional refinements and will need a thorough redesign to meet our final vision.



## Related Issue

Closes #31 , #32, #33, #35



## Acceptance Criteria
The acceptance criteria for this PR are specified in [Issue #14](https://github.com/the-collab-lab/tcl-79-smart-shopping-list/issues/14).



## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

`enhancement`, `bug fix`, `refactor`, `feature`, `style`

## Updates

### Before


<img src="https://github.com/user-attachments/assets/47ace97f-f8d6-46ea-90ec-91b17c0ad0de" alt="Screenshot 2024-10-06 at 14:39:14" />
<br/>

<img src="https://github.com/user-attachments/assets/c312b5a2-2761-43fc-8139-24874f1c34e9" alt="Screenshot 2024-10-06 at 14:05:33" width="500"/>

<img src="https://github.com/user-attachments/assets/d14cd27d-557a-4b25-91b2-ae3f123b316e" alt="Screenshot 2024-10-06 at 14:05:52" width="500"/>

<img src="https://github.com/user-attachments/assets/c756f9fa-9b36-4c76-961b-a17a7f4e4adf" alt="Screenshot 2024-10-06 at 14:06:07" width="500"/>




### After



<img src="https://github.com/user-attachments/assets/9c8d0a5e-5c10-44d0-96bb-b3e2ac4375c2" alt="Screenshot 2024-10-06 at 14:44:14" width="500"/>
<img src="https://github.com/user-attachments/assets/b5f73cbe-3344-4bb4-8545-37c4799ee518" alt="Screenshot 2024-10-06 at 14:44:50" width="500"/>

<img src="https://github.com/user-attachments/assets/670dd729-c573-4ff2-aa63-a0dbd04ab717" alt="Screenshot 2024-10-06 at 14:45:06" width="500"/>
<br/>
<img src="https://github.com/user-attachments/assets/2f47bbb1-a01f-4e7a-8034-8c81688e6f30" alt="Screenshot 2024-10-06 at 14:45:33" width="500"/>
<br/>
<img src="https://github.com/user-attachments/assets/61c3f322-7b1a-4253-8b34-b182d0aa9c18" alt="Screenshot 2024-10-06 at 14:46:00" width="300"/>
<br/>
<img src="https://github.com/user-attachments/assets/ffb1b077-e8b3-4583-98e8-6513177cfab7" alt="Screenshot 2024-10-06 at 14:46:18" width="300"/>
<br/>
<img src="https://github.com/user-attachments/assets/12195ef7-132e-4a10-9277-acb41726cd4b" alt="Screenshot 2024-10-06 at 14:46:42" width="300"/>
<br/>
<img src="https://github.com/user-attachments/assets/9e83ca82-ac07-4251-9d51-f7cc93368652" alt="Screenshot 2024-10-06 at 14:39:44" />

<br/>
<img src="https://github.com/user-attachments/assets/20483b7f-222d-46fe-8ef8-73eeca797223" alt="Screenshot 2024-10-06 at 14:40:07" width="500"/>

<img src="https://github.com/user-attachments/assets/5923d4e8-2cfb-4157-832a-e2757efc4ea7" alt="Screenshot 2024-10-06 at 14:40:26" width="500"/>

<img src="https://github.com/user-attachments/assets/e3e1a9ea-415e-471a-bf67-1e7ed907c5f9" alt="Screenshot 2024-10-06 at 14:43:46" width="500"/>




## Testing Steps / QA Criteria

After this PR is merged, ensure you're on the main branch.

- Run `git checkout main` and pull the latest changes with `git pull`.
- Run `npm install` to ensure all dependencies are up to date.
- 
After this, you can check the app's functionality and basic styling.

Note: Some styling adjustments are still pending, and the overall design will require additional refinements in future updates following today's merge.
